agent/agent_endpoint.go

```diff
@@ -88,7 +88,8 @@ func (s *HTTPServer) AgentSelf(resp http.ResponseWriter, req *http.Request) (int
 
 // enablePrometheusOutput will look for Prometheus mime-type or format Query parameter the same way as Nomad
 func enablePrometheusOutput(req *http.Request) bool {
-	if format := req.URL.Query().Get("format"); format == "prometheus" {
+	if format := req.URL.Query().Get("format"); format == "prometheus" ||
+		strings.HasPrefix(req.Header.Get("Accept"), "application/openmetrics-text;") {
```

Comment:
I think it would better to check for the string to include `application/openmetrics-text` or the old prometheus Accept header `text/plain;version=0.0.4` in order to be compatible with not so old Prometheus versions (the `application/openmetrics-text` mime was introduced less than 1 year ago)

Reply:
[`application/openmetrics-text` is used since Prometheus 2.5 in October last year](https://github.com/prometheus/prometheus/commit/9c03e11c2cf2ad6c638567471faa5c0f6c11ba3d), which is already pretty old (regarding the project's age). The documentation you linked is outdated from a perspective that OpenMetrics has its own repository, the Prometheus website isn't update either, on the other hand...

In the end I don't personally care -- I understand everybody not having good feelings with the old accept string, but I don't see it would hurt too much. Not supporting `text/plain` would not be a regression.

Reply:
Agree with plain/text and version, but you sucks at ascending compat

Reply:
October last year is pretty old....
we don't live in same world.
My daughter is pretty young and she is 7 years old ;)

Reply:
Heh I don't think it's at all intentional but the tone of this thread could be read as being a little personal!

I don't think it matters terribly whether 8 months is "old" or not. In bleeding edge green-field Kubernetes world it's pretty old but in many large Enterprises who don't upgrade key infrastructure more than once every 18 month (if you're lucky) then it's clearly not old enough.

My two cents here is that the new string seems much cleaner at stating actual intent but I don't honestly think any random HTTP service is going to be hitting a stats endpoing with a non-standard accept header that happens to look just like Prometheus' old one and be surprised to get Prometheus format back.

Part of the reason we erred away last time was that it looked generic and was different from other products but importantly there was no concrete use-case that wasn't already easily met with the `?format` param. Now we've seen a few things including popular Kube integrations that can't use that so I'd say this make sense to merge.

I'm going to check Nomad and Vault teams agree first so we don't end up all making different decisions though.

Thanks for your contribution @JensErat and thanks @pierresouchay for your continued diligence in responding to folks here.

Reply:
Looking at the accept-string again I think the check I proposed is slightly too simple; and the test needs a split for `,` first and then prefix-matching. Is there something like a shared library between the three projects? Otherwise, the function would still be rather simple, but will probably require another three lines for a loop.

If there is agreement on what to do (old/new content type) I will gladly provide pull requests for the three projects. I don't really have an opinion on including the old MIME string or not -- just needs some consensus here.

Reply:
@JensErat Sorry if my comment was offensive, by reading it again, I think it was bit too personal, I agree with @banks. That was not my intention. Deep Apologies about that @JensErat . My point was just I would be so glad to benefit from it and we are still on older versions :-)

I really think it would be great to avoid the parameters stuff in prometheus (which is non-trivial, compared to path, I had to fight with it)